src/librustc/middle/const_eval.rs

```diff
@@ -600,7 +600,7 @@ pub fn lit_to_const(lit: &ast::Lit) -> const_val {
     match lit.node {
         ast::LitStr(ref s, _) => const_str((*s).clone()),
         ast::LitBinary(ref data) => {
-            const_binary(Rc::new(data.iter().map(|x| *x).collect()))
+            const_binary(data.clone())
```

Comment:
Yeah, assumed safe rust usage which makes this equivalent? Tests passed for a full `make check` so it should be fine?